grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,518 @@
+        # (the more "wrong" cases tested the better)
+        
+        action_domains = {
+            'agent_0' : 0,
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        # above action domain should raise an error: "0" is not an iterable !
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_0")
+        
+        action_domains = {
+            'agent_0' : [0],
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        # above action domain should raise an error: substations are not fully allocated !
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_1")
+            
+        action_domains = {
+            'agent_0' : [],
+            'agent_1' : list(range(self.env.n_sub))
+        }
+        # above action domain should raise an error: agents must have at least one substation !
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_2")
+            
+        action_domains = {
+            'agent_0' : [0,1,6,3, 4],
+            'agent_1' : [5,2,7,8,9,10,11,12,13]
+        }
+        # this domain is valid even if it is not connected
+        try:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_3")
+        except DomainException:
+            self.fail("action_domains raised Domain Exception unexpectedly!")
+            
+        action_domains = {
+            'agent_0' : [0,1,6,3, 4, 5],
+            'agent_1' : [5,2,7,8,9,10,11,12,13]
+        }
+        # this domain is valid even if it is not connected
+        with self.assertRaises(DomainException) as de:
+            MultiAgentEnv(self.env, action_domains, _add_to_name="test_verify_domains_4")
+        
+    
+    def test_build_subgrids_action_domains(self):
+        """Tests that the action_domains are correctly defined 
+            in MultiAgentEnv._build_subgrid_from_domain method
+        """
+        self.ma_env = MultiAgentEnv(self.env, self.action_domains, _add_to_name="test_build_subgrids_action_domains")
+        assert self.ma_env._action_domains['agent_0']['sub_id'] == self.action_domains['agent_0']
+        assert self.ma_env._action_domains['agent_1']['sub_id'] == self.action_domains['agent_1']
+    
+    def test_masks(self):
+        # We compare the masks with known values
+        self.ma_env = MultiAgentEnv(self.env, self.action_domains, _add_to_name="test_masks")
+        assert (self.ma_env._action_domains['agent_0']['mask_load'] == [True,  True,  True,  True, False, False, False, 
+                                                                        False, False, False, False]).all()
+        assert (self.ma_env._action_domains['agent_1']['mask_load'] == np.invert([True,  True,  True,  True, False, False, 
```

Comment:
Do not copy paste the [True, True, True, True, False, False] vector.

Set it to a variable, say "mak_load_agent_0 = np.array(XXX)" then you can `assert np.all(self.ma_env._action_domains["agent_0"]["mask_load"] == mask_load_agent_0)`
and 
`assert np.all(self.ma_env._action_domains["agent_1"]["mask_load"] == ~mask_load_agent_0)`

And so it's much much clearer what you are trying to test.

Reply:
Done.